diff --git a/arch/arm/lib/Makefile b/arch/arm/lib/Makefile
index 0410b7c..ac12bd4 100644
--- a/arch/arm/lib/Makefile
+++ b/arch/arm/lib/Makefile
@@ -51,7 +51,7 @@ $(obj)/csumpartialcopy.o:	$(obj)/csumpartialcopygeneric.S
 $(obj)/csumpartialcopyuser.o:	$(obj)/csumpartialcopygeneric.S
 
 ifeq ($(CONFIG_KERNEL_MODE_NEON),y)
-  NEON_FLAGS			:= -mfloat-abi=softfp -marm -mfpu=neon-vfpv4 -funsafe-math-optimizations
+  NEON_FLAGS			:= -mfloat-abi=hard -marm -mfpu=neon-vfpv4 -funsafe-math-optimizations
   CFLAGS_xor-neon.o		+= $(NEON_FLAGS)
   obj-$(CONFIG_XOR_BLOCKS)	+= xor-neon.o
 endif
